src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1232,22 +1235,54 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
     filterVariants(p)
   }
 
-  def filterVariantsList(input: String, keep: Boolean): VariantSampleMatrix[T] = {
-    copy(
-      rdd = rdd
-        .orderedLeftJoinDistinct(Variant.variantUnitRdd(sparkContext, input).toOrderedRDD)
-        .mapPartitions({ it =>
-          it.flatMap { case (v, ((va, gs), o)) =>
-            o match {
-              case Some(_) =>
-                if (keep) Some((v, (va, gs))) else None
-              case None =>
-                if (keep) None else Some((v, (va, gs)))
-            }
-          }
-        }, preservesPartitioning = true)
-        .asOrderedRDD
-    )
+  def filterVariantsList(variants: java.util.ArrayList[Variant], keep: Boolean): VariantSampleMatrix[T] =
+    filterVariantsList(variants.asScala.toSet, keep)
+
+  def filterVariantsList(variants: Set[Variant], keep: Boolean): VariantSampleMatrix[T] = {
+    val partitionVariants = variants
+      .groupBy(v => rdd.orderedPartitioner.getPartition(v))
+      .toArray
+      .sortBy(_._1)
+
+    val adjRDD = new AdjustedPartitionsRDD[RowT](rdd,
```

Comment:
You want me to do a broadcast per partition?  That seems wrong.

My assumption was the variants per partition will be small.  That will be in the AdjustedPartitionsRDDPartition.  That will be sent out just for the task which will compute that partition.  Broadcasting all the variants to everyone seems wrong.

Ah, you're right about keep=False.  I will rework that.